linetools/spectra/xspectrum1d.py

```diff
@@ -241,28 +328,28 @@ def add_noise(self, seed=None, s2n=None):
         '''
         # Seed
         np.random.seed(seed=seed)
-        #
-        npix = len(self.flux)
+
         # Random numbers
-        rand = np.random.normal(size=npix)
+        rand = np.random.normal(size=self.npix)
 
         # Modify the flux
         if s2n is not None:
             sig = 1. / s2n
         else:
-            sig = self.sig
+            sig = self.sig.value
         #
-        self.flux = self.flux + (rand * sig) * self.flux.unit
+        self._data[self.select]['flux'] = self.flux.value + (rand * sig)
```

Comment:
Should we store the 'original' spectrum (i.e. flux values with no extra noise) somewhere (maybe something like `self._data['flux_orig']`)? I do not like the functionality of losing information without a way to recover the original one.